src/socket.cr

```diff
@@ -31,110 +31,310 @@ class Socket < IO::FileDescriptor
     INET6  = LibC::AF_INET6
   end
 
-  struct IPAddress
-    getter family : Family
-    getter address : String
-    getter port : UInt16
-
-    def initialize(@family : Family, @address : String, port : Int)
-      if family != Family::INET && family != Family::INET6
-        raise ArgumentError.new("Unsupported address family")
-      end
+  # :nodoc:
+  SOMAXCONN = 128
+
+  property family : Family
+  property type : Type
+  property protocol : Protocol
```

Comment:
Can these three be changed after creating a socket? If not, they should probably be getters.